scitt_emulator/did_helpers.py

```diff
@@ -0,0 +1,193 @@
+import os
+import sys
+import inspect
+import urllib.parse
+from typing import Optional, Callable, Dict, Tuple, Union
+
+import multibase
+import multicodec
+import cryptography.hazmat.primitives.asymmetric.ec
+
+
+def did_web_to_url(
+    did_web_string: str,
+    *,
+    scheme: Optional[str] = None,
+):
+    if scheme is None:
+        scheme = os.environ.get("DID_WEB_ASSUME_SCHEME", "https")
+    return "/".join(
+        [
+            f"{scheme}:/",
+            *[urllib.parse.unquote(i) for i in did_web_string.split(":")[2:]],
+        ]
+    )
+
+
+class DIDKeyInvalidPublicKeyLengthError(ValueError):
+    """
+    If the byte length of rawPublicKeyBytes does not match the expected public
+    key length for the associated multicodecValue, an invalidPublicKeyLength
+    error MUST be raised.
+    """
+
+
+class DIDKeyDecoderNotFoundError(NotImplementedError):
+    """
+    Raised when we don't have a function implemented to decode the given key
+    """
+
+
+class DIDKeyDecoderError(Exception):
+    """
+    Raised when we failed to decode a key from a did:key DID method
+    """
+
+
+class DIDKeyInvalidPublicKeyError(DIDKeyDecoderError):
+    """
+    Raised when the raw bytes of a key are invalid during decode
+    """
+
+
+DID_KEY_METHOD = "did:key:"
```

Comment:
I would not use did:key since it relies on MultiBase, which is not a standard.  See my related comment.